### ISSUE TITLE
Copying strided slices does not work on structured arrays with MD  fields (migrated from Trac #1760)

### ISSUE THREAD

**Issue by thouis** [thouis/numpy-trac-migration#3310]
Original ticket http://projects.scipy.org/numpy/ticket/1760
Reported 2011-03-07 by atmention:FrancescAlted, assigned to unknown.

The next code snippet shows the problem:

```
import numpy as np

print "numpy version:", np.__version__

r=np.fromstring('abcd'*16*5, dtype='i4,(2,3)u2')
#start, stop, step = 0, 5, 1    # step == 1 works
start, stop, step = 0, 5, 2
r2 = r[start:stop:step]
print "r -->", r[start]['f1']
print "r2 -->", r2[0]['f1']
assert np.all(r[start]['f1'] == r2[0]['f1'])
```

Using numpy 1.5.x:

```
numpy version: 1.5.0.dev
r --> [[25185 25699 25185]
 [25699 25185 25699]]
r2 --> [[25185 25699 25185]
 [25699 25185 25699]]
```

Using current 1.6 in git master:

```
numpy version: 1.6.0.dev-c081ad7
r --> [[25185 25699 25185]
 [25699 25185 25699]]
r2 --> [[25185 25185 25185]
 [25699 25699 25699]]
Traceback (most recent call last):
  File "/tmp/copy-recarr.py", line 12, in <module>
    assert np.all(r[start]['f1'] == r2[0]['f1'])
AssertionError
```

**Comment by thouis** [comment]
Comment in Trac by atmention:mwiebe, 2011-03-07

Fixed in b28b1158fc58459b80b2.

**Comment by thouis** [comment]
Comment in Trac by atmention:mwiebe, 2011-05-31
